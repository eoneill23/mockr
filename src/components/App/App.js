import React, { useState, useMemo } from 'react';
import HomePage from '../HomePage/HomePage';
import NavBar from '../NavBar/NavBar';
import LoginForm from '../LoginForm/LoginForm';
import Dashboard from '../Dashboard/Dashboard';
import StudentInterviewContainer from '../StudentInterviewContainer/StudentInterviewContainer';
import AdminSearchStudentContainer from '../AdminSearchStudentContainer/AdminSearchStudentContainer';
import InterviewSetup from '../InterviewSetup/InterviewSetup';
import AdminAllQuestionsContainer from '../AdminAllQuestionsContainer/AdminAllQuestionsContainer';
import { InterviewsContext, UserContext, QuestionsContext } from '../../Context';
import { Route, Redirect } from 'react-router-dom';
import Interview from '../Interview/Interview';
import StudentQuestions from '../StudentQuestions/StudentQuestions';

// Apollo
import ApolloClient from 'apollo-boost';
import {ApolloProvider} from '@apollo/react-hooks';
const client = new ApolloClient({
  uri: 'https://thawing-wave-76846.herokuapp.com/graphql'
});

export const App = () => {
  const [user, setUser] = useState('');
  const [interviews, setInterviews] = useState([]);
  const [questions, setQuestions] = useState([]);

  const userInfo = useMemo(() => ({ user, setUser }), [user, setUser]);
  const fetchedInterviews = useMemo(() => ({ interviews, setInterviews }), [interviews, setInterviews]);
  const fetchedQuestions = useMemo(() => ({ questions, setQuestions }), [questions, setQuestions]);

  return (
    <ApolloProvider client={client}>
      <main className='main'>
        <QuestionsContext.Provider value={fetchedQuestions}>
          <InterviewsContext.Provider value={fetchedInterviews}>
            <UserContext.Provider value={userInfo}>
              <NavBar />
              <Route exact path='/'><HomePage /></Route>
              <Route exact path='/login' render={() => user ? (<Redirect to='/dashboard'/>) : <LoginForm />}/>
              <Route exact path='/dashboard' render={() => !user ? (<Redirect to='/login'/>) : <Dashboard /> } />
              <Route exact path='/student-interviews' render={() => !user ? (<Redirect to='/login'/>) : <StudentInterviewContainer /> } />
              <Route exact path='/student-questions' render={() => !user ? (<Redirect to='/login'/>) : <StudentQuestionContainer /> } />
              <Route exact path='/interview' render={() => !user.currentInterview ? (<Redirect to='/login'/>) : <Interview /> } />
              <Route exact path='/students' render={() => user.role !== 2 ? (<Redirect to='/login'/>) : <AdminSearchStudentContainer />} />
              <Route exact path='/all-questions' render={() => !user ? (<Redirect to='/login'/>) : <AdminAllQuestionsContainer /> } />
              <Route exact path='/select-student' render={() => user.currentInterview ? (<Redirect to='/interview'/>) : <InterviewSetup />}/>
            </UserContext.Provider>
          </InterviewsContext.Provider>
        </QuestionsContext.Provider>
      </main>
    </ApolloProvider>
  );
}

export default App;
