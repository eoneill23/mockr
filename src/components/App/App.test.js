import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router';
import { render } from '@testing-library/react';
import App from './App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(
      <MemoryRouter intialEntries={['/']}>
        <App />
      </MemoryRouter>
    );
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
