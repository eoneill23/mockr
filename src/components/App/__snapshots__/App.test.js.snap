// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
      >
        <nav
          class="nav"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Mockr
          </a>
           
          <a
            href="/login"
          >
            Log In
          </a>
          
        </nav>
        <section
          class="welcome-info"
        >
          <p>
            Welcome to Mockr! Mockr is an app that facilitates mock interviews between interviewers and interviewees. Its intuitive design allows an interviewer to ask questions commonly asked in an interview and rate the interviewee's response on a scale from 1-5.
          </p>
          <p>
            It also allows interviewers to email interviewees with their notes from the interview to provide a record of their feedback.
          </p>
          <p>
            Click log in at the top right corner of the page to log in and begin the mock interview process!
          </p>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
    >
      <nav
        class="nav"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Mockr
        </a>
         
        <a
          href="/login"
        >
          Log In
        </a>
        
      </nav>
      <section
        class="welcome-info"
      >
        <p>
          Welcome to Mockr! Mockr is an app that facilitates mock interviews between interviewers and interviewees. Its intuitive design allows an interviewer to ask questions commonly asked in an interview and rate the interviewee's response on a scale from 1-5.
        </p>
        <p>
          It also allows interviewers to email interviewees with their notes from the interview to provide a record of their feedback.
        </p>
        <p>
          Click log in at the top right corner of the page to log in and begin the mock interview process!
        </p>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
